beam-core/Database/Beam/Query/Combinators.hs

```diff
@@ -112,11 +112,17 @@ values_ :: forall be db s a
            , BeamSqlBackend be )
         => [ a ] -> Q be db s a
 values_ rows =
-    Q $ liftF (QAll (\tblPfx -> fromTable (tableFromValues (map (\row -> project (Proxy @be) row tblPfx) rows)) . Just . (,Just fieldNames))
-                    (\tblNm' -> fst $ mkFieldNames (qualifiedField tblNm'))
+    Q $ liftF (QAll (\tblPfx -> fromTable (tableFromValues colCount (map (\row -> project (Proxy @be) row tblPfx) rows)) . Just . (,colAliases))
+                    (\tblNm' -> if useAliases
+                                then fst $ mkFieldNames (qualifiedField tblNm')
+                                else fst $ mkDefaultFieldNames (qualifiedField tblNm'))
                     (\_ -> Nothing) snd)
     where
+      useAliases = knownBool @(BeamSqlBackendSupportsColumnAliases be)
+      colAliases | useAliases = Just fieldNames
```

Comment:
Is there a reason to use type-level bools for this? Seems like a typeclass method using either `mkFieldNames` or `mkDefaultFieldNames` might be more straightforward

Reply:
Good point.. yeah that would be easier